src/cascadia/TerminalCore/TerminalApi.cpp

```diff
@@ -560,3 +560,31 @@ try
     return true;
 }
 CATCH_LOG_RETURN_FALSE()
+
+// Method Description:
+// - Updates the buffer's current text attributes depending on whether we are
+//   starting/ending a hyperlink
+// Arguments:
+// - The hyperlink URI
+// Return Value:
+// - true
+bool Terminal::AddHyperlink(std::wstring_view uri, std::wstring_view params) noexcept
+{
+    auto attr = _buffer->GetCurrentAttributes();
+    if (uri.empty())
+    {
+        // URI is empty, this means we are ending a hyperlink
+        attr.SetHyperlinkId(0);
+        _buffer->SetCurrentAttributes(attr);
+    }
+    else
+    {
+        // URI is non-empty, this means we are starting a hyperlink
+        // Conpty will always send us an id in params for non-empty URIs, parse it out here
+        const auto id = (USHORT)_wcstoui64(params.data(), NULL, 10);
+        attr.SetHyperlinkId(id);
+        _buffer->SetCurrentAttributes(attr);
+        _buffer->AddHyperlinkToMap(uri, id);
+    }
+    return true;
+}

```

Comment:
This might just be a personal preference, but I think this would also make more sense split up into two methods `StartHyperlink(URI, params)` and `EndHyperlink`. I know that's a lot more copy/pasta for the function signatures through all the interfaces, but I think that might help make it clearer than just "pass \"\" to `AddHyperlink` to end the hyperlink".

Reply:
Gonna resolve this and leave the comment thread above for where we discuss this